### PR TITLE
Remove ring, fix clippy warnings and migrate to Rust 2018

### PR DESCRIPTION
**Pull request by kigawas**

### CODE REVIEW

**Comment by kigawas** [comment]
https://github.com/mortendahl/rust-paillier/pull/68

File: Cargo.toml [hunk 1]
```diff
@@ -1,25 +1,20 @@
...
-ring = { version="0.13", optional=true }
-
-bit-vec = "0.5"
+num-traits = "0.2"
+rand = "0.6"
+rayon = "1.1"
+serde =  { version = "1.0", features = ["derive"] }
+bit-vec = "0.6"
+rust-gmp = { version = "0.5", optional=true }
+rust-crypto = { version = "^0.2", optional = true }
```
Comment (omershlo) [inline Cargo.toml]: why this library is needed here?

File: Cargo.toml [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
-rust-gmp = { version="0.5", optional=true }
-ring = { version="0.13", optional=true }
-
-bit-vec = "0.5"
+num-traits = "0.2"
+rand = "0.6"
+rayon = "1.1"
+serde =  { version = "1.0", features = ["derive"] }
+bit-vec = "0.6"
+rust-gmp = { version = "0.5", optional=true }
```
Comment (omershlo) [inline Cargo.toml]: based on #1 it looks that this library can be removed safely

**Comment by omershlo** [comment]
Can you make sure #1 changes is also included in this PR?

File: Cargo.toml (hunk 2 above)
Comment (kigawas) [reply Cargo.toml]: no problem

**Pull request merged**